### ISSUE TITLE
Compilation error: "error: namespace "at::cuda" has no member "getCurrentCUDAStream""

### ISSUE THREAD

**Issue by yuxuan-z19** [cuhk-eda/Xplace#1]
Compilation of `neural` fails while using both PyTorch 1.12.1 with CUDA 11.6,  PyTorch Nightly with CUDA 11.7

```bash
/home/zyx/eda-prj/Xplace/cpp_to_py/density_map_cuda/density_map_cuda_kernel.cu(155): error: namespace "at::cuda" has no member "getCurrentCUDAStream"

/home/zyx/eda-prj/Xplace/cpp_to_py/density_map_cuda/density_map_cuda_kernel.cu(183): error: namespace "at::cuda" has no member "getCurrentCUDAStream"

/home/zyx/eda-prj/Xplace/cpp_to_py/density_map_cuda/density_map_cuda_kernel.cu(211): error: namespace "at::cuda" has no member "getCurrentCUDAStream"

/home/zyx/eda-prj/Xplace/cpp_to_py/wa_wirelength_hpwl_cuda/wa_wirelength_hpwl_cuda_kernel.cu(265): error: namespace "at::cuda" has no member "getCurrentCUDAStream"

/home/zyx/eda-prj/Xplace/cpp_to_py/wa_wirelength_hpwl_cuda/wa_wirelength_hpwl_cuda_kernel.cu(318): error: namespace "at::cuda" has no member "getCurrentCUDAStream"

/home/zyx/eda-prj/Xplace/cpp_to_py/density_map_cuda/density_map_cuda_naive_kernel.cu(152): error: namespace "at::cuda" has no member "getDefaultCUDAStream"

/home/zyx/eda-prj/Xplace/cpp_to_py/wa_wirelength_hpwl_cuda/wa_wirelength_hpwl_cuda_kernel.cu(365): error: namespace "at::cuda" has no member "getCurrentCUDAStream"

/home/zyx/eda-prj/Xplace/cpp_to_py/density_map_cuda/density_map_cuda_naive_kernel.cu(192): error: namespace "at::cuda" has no member "getDefaultCUDAStream"

/home/zyx/eda-prj/Xplace/cpp_to_py/wa_wirelength_hpwl_cuda/wa_wirelength_hpwl_cuda_kernel.cu(421): error: namespace "at::cuda" has no member "getCurrentCUDAStream"
```

CMake:
```bash
-- The C compiler identification is GNU 9.4.0
-- The CXX compiler identification is GNU 9.4.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- CMAKE_BUILD_TYPE: Release
-- PROJECT_SOURCE_DIR=/home/zyx/eda-prj/Xplace
-- CMAKE_CXX_ABI: _GLIBCXX_USE_CXX11_ABI=0
-- pybind11 v2.11.0 dev1
-- Found PythonInterp: /home/zyx/eda-prj/Xplace/.venv/bin/python (found suitable version "3.8.10", minimum required is "3.6") 
-- Found PythonLibs: /usr/lib/x86_64-linux-gnu/libpython3.8.so
-- Performing Test HAS_FLTO
-- Performing Test HAS_FLTO - Success
-- PYTHON_INCLUDE_DIRS: /usr/include/python3.8
-- FLUTE_INCLUDE_DIR: /home/zyx/eda-prj/Xplace/thirdparty/flute
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.1") 
-- Checking for module 'cairo'
--   Found cairo, version 1.16.0
-- Found Cairo: /usr/include/cairo  
-- CAIRO_INCLUDE_DIRS: /usr/include/cairo
-- CAIRO_LIBRARIES: /usr/lib/x86_64-linux-gnu/libcairo.so
-- TORCH_INSTALL_PREFIX=/home/zyx/eda-prj/Xplace/.venv/lib/python3.8/site-packages/torch
-- TORCH_VERSION=1.12
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Found CUDA: /usr/local/cuda-11.7 (found suitable version "11.7", minimum required is "11.0") 
-- TORCH_ENABLE_CUDA=1
-- CUDA_ARCH_FLAGS: -gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86
-- CUDA_NVCC_FLAGS: -gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86;--compiler-options;-fPIC;-std=c++17;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86;--extended-lambda;-D__CUDA_NO_HALF_OPERATORS__;-D__CUDA_NO_HALF_CONVERSIONS__;-D__CUDA_NO_BFLOAT16_CONVERSIONS__;-D__CUDA_NO_HALF2_OPERATORS__;--expt-relaxed-constexpr
-- TORCH_INCLUDE_DIRS=/usr/include/python3.8/home/zyx/eda-prj/Xplace/.venv/lib/python3.8/site-packages/torch/include/home/zyx/eda-prj/Xplace/.venv/lib/python3.8/site-packages/torch/include/torch/csrc/api/include
-- Configuring done
-- Generating done
-- Build files have been written to: /home/zyx/eda-prj/Xplace/build
```

**Comment by liulixinkerry** [comment]
@yuxuan-z19 Thanks for your issue. Could you please try PyTorch 1.10?

**Comment by liulixinkerry** [comment]
@yuxuan-z19 
Thanks for your issue. We just fixed it by including the library needed by PyTorch 1.12. Please pull the newest commit.

**Comment by yuxuan-z19** [comment]
bug is fixed on the `main` branch with PyTorch 1.12.1 with CUDA 11.6; @liulixinkerry perhaps a git merge is needed for the `neural` branch. Closing this issue.

**Comment by liulixinkerry** [comment]
Merged. Thanks a lot.
